lib/src/fix.rs

```diff
@@ -291,11 +307,20 @@ pub fn fix_files(
                         file_id: id.clone(),
                         repo_path: repo_path.clone(),
                     };
-                    if let Some(new_id) = fixed_file_ids.get(&file_to_fix) {
-                        return Some(TreeValue::File {
-                            id: new_id.clone(),
-                            executable: *executable,
-                        });
+                    if let Some(fix_result) = fix_results.get(&file_to_fix) {
+                        if fix_result.file_id != file_to_fix.file_id {
+                            return Some(TreeValue::File {
+                                id: fix_result.file_id.clone(),
+                                executable: *executable,
+                            });
+                        } else {
+                            // TODO: consider counting files examined but
+                            // unchanged in FixSummary.
+                        }
+                        if !fix_result.messages.is_empty() {
+                            // TODO: Add messages to FixSummary (but put an
+                            // upper bound on this?).
```

Comment:
How do you want to achieve that? Since I think it relates to `clang-format` spewing to many errors, I think @hooper also can chime in.

Reply:
I addressed a couple of the comments, while leaving some unresolved.

In particular, I am not sure how I wound put an upper bound on fix messages.

Please note I like the much simpler approach of https://github.com/jj-vcs/jj/pull/6226. I don't mind withdrawing this PR in favor of that one.

Reply:
I addressed a couple of the comments, while leaving some unresolved.

In particular, I am not sure how I wound put an upper bound on fix messages.

Please note I like the much simpler approach of https://github.com/jj-vcs/jj/pull/6226. I don't mind withdrawing this PR in favor of that one.